internal/allocator/static_allocator.go

```diff
@@ -0,0 +1,58 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package allocator
+
+import (
+	"fmt"
+	"sync"
+)
+
+// staticAllocator implements the Interface.
+// It is constructed from a range of IDs.
+// Once all IDs are allocated, an error will be returned.
+type staticAllocator struct {
```

Comment:
We could utilize a map[segmentID, latestLogID] instead of the staticAllocator. Using a map provides the following benefits:
1. LogID can begin from 0 for each segment.
2. No reliance on the Allocator interface is necessary.

Reply:
SyncManager is used not only for import but also for streaming flush. Since SyncManager relies on the Allocator interface, staticAllocator is necessary.

Reply:
Would it be better to use localAllocator instead of staticAllocator?

Reply:
You mean renaming it to localAllocator?